android/src/com/mapswithme/maps/widget/placepage/SponsoredHotel.java

```diff
@@ -73,6 +173,15 @@ static void requestPrice(String id, String currencyCode)
     nativeRequestPrice(id, currencyCode);
   }
 
+  static void requestInfo(String id, String locale)
+  {
+    HotelInfo info = sInfoCache.get(id);
+    if (info != null)
+      onInfoReceived(id, info);
+
+    nativeRequestInfo(id, locale);
```

Comment:
onInfoReceived is called from the C++ code that will result from a query. If there is a cache once I call onInfoReceived method (so that the user immediately saw this data) and in parallel sends a request for new data.